hydrolib/core/io/ini/util.py

```diff
@@ -58,6 +58,40 @@ def split(v: Any):
     return validator(*field_name, allow_reuse=True, pre=True)(split)
 
 
+def make_list_length_root_validator(
+    *field_names, length_name: str, length_incr: int = 0, id_name: str
+):
+    """
+    Get a root_validator that checks the correct length of several list fields in an object.
+
+    Args:
+        *field_names (str): names of the instance variables that are a list and need checking.
+        length_name (str): name of the instance variable that stores the expected length.
+        length_incr (int): Optional extra increment of length value (e.g., to have +1 extra value in lists).
+        id_name (str): name of the ``id`` instance variable that will be printed in an error message string.
+    """
+
+    def validate_correct_length(cls, values):
+        length = values.get(length_name)
+        if length is None:
+            # length attribute not present, possibly defer validation to a subclass.
+            return values
+
+        length = length + length_incr
+
+        for field_name in field_names:
+            field = values.get(field_name)
+            if field is not None and not (
+                isinstance(field, list) and len(field) == length
+            ):
+                raise ValueError(
+                    f"Number of values for {field_name} should be equal to the {length_name} value ({id_name}={values[id_name]})."
```

Comment:
This message is technically not correct when the `length_incr` is greater than 0.

Reply:
Fixed in an upcoming commit.